sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -45,7 +45,8 @@ class UnresolvedException(function: String)
 case class UnresolvedRelation(
     multipartIdentifier: Seq[String],
     options: CaseInsensitiveStringMap = CaseInsensitiveStringMap.empty(),
-    override val isStreaming: Boolean = false)
+    override val isStreaming: Boolean = false,
+    lookupError: Option[Throwable] = None)
```

Comment:
I might have missing context here: why we only add `lookupError` for relation but not on other function/namespace, etc.?

Reply:
See https://github.com/apache/spark/pull/38039/files#r987631927

I think we can remove this as well once we make `parquer`, `json` etc. as special catalogs. Then we don't need the rule `ResolveSQLOnFile` and can fail fast for `UnresolvedRelation`